client/client-multi/client-c/src/kaa/kaa_logging.c

```diff
@@ -153,24 +154,34 @@ kaa_error_t kaa_extension_logging_server_sync(void *context, uint32_t request_id
 
 kaa_error_t kaa_logging_need_logging_resync(kaa_log_collector_t *self, bool *result)
 {
-    KAA_RETURN_IF_NIL2(self, result, KAA_ERR_BADPARAM);
+    // TODO(KAA-982): Use asserts
+    if (!self || !result) {
+        return KAA_ERR_BADPARAM;
+    }
+
     if (self->is_sync_ignored) {
         *result = false;
         return KAA_ERR_NONE;
     }
+
     *result = true;
     return KAA_ERR_NONE;
 }
 
 static kaa_error_t remember_request(kaa_log_collector_t *self, uint16_t bucket_id, uint16_t count)
 {
-    KAA_RETURN_IF_NIL(self, KAA_ERR_BADPARAM);
+    // TODO(KAA-982): Use asserts
+    if (!self) {
```

Comment:
This is the supporting refactoring. We going to banish KAA_RETURN_* macro, establish error handing approach and use debug asserts for most of places, especially checks of arguments inside internal functions.